ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py

```diff
@@ -271,9 +279,14 @@ def compute_gradient_magnitude(
             use_vail_noise = True
             z_mu = self._z_mu_layer(hidden)
             hidden = torch.normal(z_mu, self._z_sigma * use_vail_noise)
-        estimate = self._estimator(hidden).squeeze(1).sum()
-        gradient = torch.autograd.grad(estimate, encoder_input, create_graph=True)[0]
+        estimate = self._estimator(hidden).squeeze(1)
+        gradient = torch.autograd.grad(
+            estimate,
+            encoder_input,
```

Comment:
I do think it makes more sense to compute the gradient with respect to the normalized observations rather than the raw ones. Not sure at all how to do this without having to take the normalization out of the NetworkBody.

Reply:
I think without doing this it's actually broken and doesn't train. 

We could do the ugly thing of reaching into the NetworkBody and grabbing the normalized obs, or grabbing the normalizer and applying it to the observation.